app/src/main/java/com/sdp13epfl2021/projmag/activities/ProjectInformationActivity.kt

```diff
@@ -327,19 +336,27 @@ class ProjectInformationActivity : AppCompatActivity() {
 
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
         if (item.itemId == R.id.shareButton) {
-            val dynamicLink = Firebase.dynamicLinks.dynamicLink {
-                link = Uri.parse("https://www.example.com/projectid=" + projectVar.id)
-                domainUriPrefix = "https://projmag.page.link/"
-                androidParameters {}
-            }
-            val linkToSend = dynamicLink.uri
+            val linkToSend = getDynamicLink()
 
             val sendIntent = Intent(Intent.ACTION_SEND)
             sendIntent.putExtra(Intent.EXTRA_TEXT, linkToSend.toString())
             sendIntent.type = "text/plain"
             startActivity(sendIntent)
             return true
         }
+        else if(item.itemId == R.id.generateQRCodeButton) {
+            val linkToSend = getDynamicLink()
+
+            val qrImage = QRCode.from(linkToSend.toString()).withSize(800,800)
+            val stream = ByteArrayOutputStream()
+            qrImage.bitmap().compress(Bitmap.CompressFormat.PNG, 100, stream)
+            val byteArray: ByteArray = stream.toByteArray()
+
+            val intent = Intent(this, QRCodeActivity::class.java)
+            intent.putExtra("qrcode",byteArray)
```

Comment:
Even if the byte array is small (~8KB), maybe moving the qr generation to QRCodeActivity and only sending the url in the the intent, could be better.

Reply:
I agree this could be better but this would be a optimization that would not give any benefit because as you have said the byte array is small we can add that to the improvements column on the project board